### PR TITLE
fix: fix multimodal example

### PR DESCRIPTION
**Pull request by bwanglzu**
- [x] Fix bugs

- [x] Add tests

- [x] Update to 0.9

- [x] Switch to Jina types

- [x] Update readme

- [x] Update copyrights

- [x] working Dockerfile

- [x] Jina Hub image

- [x] Jina Flow v1

Bug fixing includes:
1. Clean up docker images (move to Jina hub). (DONE)
2. Replace `_merge_all` with `_merge_chuinks`. (DONE)
3. Adapt changes in Jina Python client. (DONE)
4. Missing URL field while querying (DONE)

final test: worked locally

### CODE REVIEW

File: multimodal-search-tirg/flow-query.yml [hunk 1]
```diff
@@ -2,7 +2,6 @@
 with:
   prefetch: 10
   read_only: true  # better add this in the query time
-  # rest_api: true
   port_expose: $JINA_PORT
 pods:
```
Comment (JoanFM) [inline multimodal-search-tirg/flow-query.yml:6]: please update to `version 1` version of Flow

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: multimodal-search-tirg/flow-query.yml [hunk 2]
```diff
@@ -1,33 +1,34 @@
...
+  - name: encode
     uses: docker://jinaai/hub.executors.encoders.multimodal.tirg
     uses_internal: pods/encode-tirg-multimodal.yml
     # uses: pods/encode-tirg-multimodal-dummy.yml
     parallel: $PARALLEL
     timeout_ready: 600000
     uses_before: _merge_chunks
     needs: [normalize, pass_text]
-  vec_idx:
+  - name: vec_idx
```
Comment (JoanFM) [inline multimodal-search-tirg/flow-query.yml:26]: can u add `polling: all` here?

File: multimodal-search-tirg/flow-query.yml [hunk 3]
```diff
@@ -1,33 +1,34 @@
...
     uses: pods/vec.yml
     shards: $SHARDS
     timeout_ready: 100000 # larger timeout as in query time will read all the data
     separated_workspace: true
-  doc_idx:
+  - name: doc_idx
```
Comment (JoanFM) [inline multimodal-search-tirg/flow-query.yml:31]: and here

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: multimodal-search-tirg/README.MD [hunk 4]
```diff
@@ -58,13 +58,13 @@ bash ./get_data.sh data/
 Alternatively, You can [Download](https://drive.google.com/drive/folders/0B4Eo9mft9jwoamlYWFZBSHFzV3c) and extract the data from google drive.
 
 ## Build encoder images
-This will download the TIRG model + code and build the image
+This will download the TIRG model + code and build the image from [Jina-hub](https://github.com/jina-ai/jina-hub).
 ### TIRG image encoder
-Start the Docker image. Go to the TirgImageEncoder folder and run the command below: 
+Start the Docker image. Go to the [TirgImageEncoder](https://github.com/jina-ai/jina-hub/tree/master/encoders/image/TirgImageEncoder) folder and run the command below:
```
[the remark this answers is not in the document]
Comment (bwanglzu) [reply multimodal-search-tirg/README.MD]: updated

File: multimodal-search-tirg/flow-query.yml (hunk 1 above)
Comment (bwanglzu) [reply multimodal-search-tirg/flow-query.yml:7]: done

File: multimodal-search-tirg/flow-query.yml (hunk 3 above)
Comment (bwanglzu) [reply multimodal-search-tirg/flow-query.yml:32]: both updated

File: multimodal-search-tirg/README.MD [hunk 5]
```diff
@@ -57,16 +57,23 @@ bash ./get_data.sh data/
...
-## Build encoder images
-This will download the TIRG model + code and build the image from [Jina-hub](https://github.com/jina-ai/jina-hub).
+## Build encoder images from Jina hub
+To make it easier for the user, we have built and published the Docker image with the indexed documents. You can retrieve the docker image using:
 ### TIRG image encoder
-Start the Docker image. Go to the [TirgImageEncoder](https://github.com/jina-ai/jina-hub/tree/master/encoders/image/TirgImageEncoder) folder and run the command below: 
+Pull & To start the `TirgImageEncoder` Docker image, run the command below: 
+
+```bash
+docker pull jinaai/hub.executors.encoders.image.tirg:latest
```
Comment (JoanFM) [inline multimodal-search-tirg/README.MD:66]: there is no latest currently

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: multimodal-search-tirg/README.MD [hunk 6]
```diff
@@ -57,16 +57,23 @@ bash ./get_data.sh data/
 
 Alternatively, You can [Download](https://drive.google.com/drive/folders/0B4Eo9mft9jwoamlYWFZBSHFzV3c) and extract the data from google drive.
 
-## Build encoder images
-This will download the TIRG model + code and build the image
+## Build encoder images from Jina hub
```
Comment (florian-hoenicke) [inline multimodal-search-tirg/README.MD:60]: Do we actually still need to pull and run it explicitly?
I thought it is enough to specify `uses: docker://jinahub/pod.encoder.tirgimageencoder:0.0.4-0.9.17` in the `flow-index.yml`

Comment (bwanglzu) [reply multimodal-search-tirg/README.MD:60]: ah that's good to know, thanks

**Comment by florian-hoenicke** [comment]
I somehow got an exception when running it locally:
I downloaded the data, pulled and the docker images, and ran them in separated terminals.
Then I ran `python app.py --task index -n 1000 -overwrite True`
But it failed with:
`AttributeError("'function' object has no attribute 'run'") during <bound method ContainerRuntime.setup of <jina.peapods.runtimes.container.ContainerRuntime object at 0x7ff8704a6b10>>add "--show-exc-info" to see the exception stack in details`
Am I doing something wrong?

**Review by JoanFM — DISMISSED** [review]

**Comment by bwanglzu** [comment]
Had a discussion with @florian-hoenicke , tested again with a clean setup locally it worked. Florian is testing on his laoptop.

File: multimodal-search-tirg/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,4 @@
+click
```
Comment (cristianmtr) [inline multimodal-search-tirg/requirements.txt:1]: Specific versions

**Comment by florian-hoenicke** [comment]
I tested it locally and it works fine

**Review by florian-hoenicke — APPROVED** [review]

**Pull request merged**